File: backend/app/schemas/case_management.py
from pydantic import BaseModel, ConfigDict, Field


class ManagedCaseItem(BaseModel):
    id: int
    code: str
    name: str
    typeText: str
    priorityText: str
    executionStatus: str
    executionStatusText: str
    lastRunAt: str
    stepsText: str


class ManagedRequirementItem(BaseModel):
    id: str
    code: str
    title: str
    status: str
    statusText: str
    priority: str
    priorityText: str
    caseCount: int
    owner: str
    updatedAt: str


class RequirementListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    requirements: list[ManagedRequirementItem]


class RequirementDetailResponse(BaseModel):
    code: str
    title: str
    status: str
    statusText: str
    priorityText: str
    owner: str
    cases: list[ManagedCaseItem]


class CreateManagedCaseInput(BaseModel):
    code: str
    name: str
    priorityText: str = "P1"
    stepsText: str = ""


class CreateRequirementRequest(BaseModel):
    code: str
    title: str
    owner: str = "当前用户"
    priority: str = "p1"
    cases: list[CreateManagedCaseInput]


class UpdateRequirementRequest(BaseModel):
    code: str
    title: str
    owner: str | None = None


class UpdatePriorityRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)
    priority: str = Field(..., description="p0 / p1 / p2")


class UpdateCaseExecutionRequest(BaseModel):
    execution_status: str


class CodeExistsResponse(BaseModel):
    exists: bool

File: frontend/src/api/caseManagement.ts
import { http } from "./http";

export type RequirementProgressStatus = "not_started" | "in_progress" | "completed";

export type RequirementItem = {
  id: string;
  code: string;
  title: string;
  status: RequirementProgressStatus;
  statusText: string;
  priority: string;
  priorityText: string;
  caseCount: number;
  owner: string;
  updatedAt: string;
};

export type CaseExecutionStatus = "not_executed" | "success" | "failed" | "blocked";

export type RequirementDetail = {
  code: string;
  title: string;
  status: RequirementProgressStatus;
  statusText: string;
  priorityText: string;
  owner: string;
  cases: Array<{
    id: number;
    code: string;
    name: string;
    typeText: string;
    priorityText: string;
    executionStatus: CaseExecutionStatus;
    executionStatusText: string;
    lastRunAt: string;
    stepsText: string;
  }>;
};

export type RequirementListResponse = {
  total: number;
  page: number;
  page_size: number;
  requirements: RequirementItem[];
};

export async function listRequirementsApi(
  page = 1,
  pageSize = 10,
  keyword?: string
): Promise<RequirementListResponse> {
  const resp = await http.get<RequirementListResponse>("/api/v1/case-management/requirements", {
    params: { page, page_size: pageSize, q: keyword?.trim() || undefined }
  });
  return resp.data;
}

export async function getRequirementDetailApi(reqCode: string): Promise<RequirementDetail> {
  const resp = await http.get<RequirementDetail>(`/api/v1/case-management/requirements/${reqCode}`);
  return resp.data;
}

export async function checkRequirementCodeExistsApi(
  code: string,
  excludeCode?: string
): Promise<{ exists: boolean }> {
  const resp = await http.get<{ exists: boolean }>("/api/v1/case-management/requirements/code-exists", {
    params: { code, exclude_code: excludeCode }
  });
  return resp.data;
}

export async function createRequirementApi(payload: {
  code: string;
  title: string;
  owner?: string;
  priority: string;
  cases: Array<{ code: string; name: string; priorityText: string; stepsText: string }>;
}): Promise<{ ok: boolean }> {
  const resp = await http.post<{ ok: boolean }>("/api/v1/case-management/requirements", payload);
  return resp.data;
}

export async function updateRequirementApi(
  reqCode: string,
  payload: { code: string; title: string }
): Promise<{ ok: boolean }> {
  const resp = await http.put<{ ok: boolean }>(`/api/v1/case-management/requirements/${reqCode}`, payload);
  return resp.data;
}

export async function updateRequirementPriorityApi(reqCode: string, priority: string): Promise<{ ok: boolean }> {
  const resp = await http.patch<{ ok: boolean }>(`/api/v1/case-management/requirements/${reqCode}/priority`, {
    priority
  });
  return resp.data;
}

export async function updateCaseExecutionApi(
  reqCode: string,
  caseId: number,
  executionStatus: CaseExecutionStatus
): Promise<{ ok: boolean }> {
  const code = encodeURIComponent(reqCode);
  const resp = await http.patch<{ ok: boolean }>(
    `/api/v1/case-management/requirements/${code}/cases/${caseId}/execution`,
    { execution_status: executionStatus }
  );
  return resp.data;
}

export async function deleteRequirementApi(reqCode: string): Promise<{ ok: boolean }> {
  const resp = await http.delete<{ ok: boolean }>(`/api/v1/case-management/requirements/${reqCode}`);
  return resp.data;
}
